src/HubPortal.sol

```diff
@@ -83,6 +91,40 @@ contract HubPortal is IHubPortal, Portal {
         emit RegistrarListStatusSent(destinationChainId_, messageId_, listName_, account_, status_);
     }
 
+    /// @inheritdoc IHubPortal
+    function sendMerkleRoots(bytes32 refundAddress_) external payable returns (uint64 sequence_) {
+        bytes32 destinationToken_ = destinationMToken[_SOLANA_WORMHOLE_CHAIN_ID];
+        IMerkleTreeBuilder merkleTreeBuilder_ = IMerkleTreeBuilder(merkleTreeBuilder);
+        bytes32 earnersMerkleRoot = merkleTreeBuilder_.getRoot(_SOLANA_EARNER_LIST);
+        bytes32 earnManagersMerkleRoot = merkleTreeBuilder_.getRoot(_SOLANA_EARN_MANAGER_LIST);
+
+        bytes memory additionalPayload_ = PayloadEncoder.encodeAdditionalPayload(
+            _currentIndex(),
+            destinationToken_,
+            earnersMerkleRoot,
+            earnManagersMerkleRoot
+        );
+
+        sequence_ = _transferNativeToken(
+            0,
+            token,
+            _SOLANA_WORMHOLE_CHAIN_ID,
+            destinationToken_,
+            refundAddress_, // recipient doesn't matter since transfer amount is 0
+            refundAddress_,
+            additionalPayload_
+        );
+
+        emit MerkleRootsSent(earnersMerkleRoot, earnManagersMerkleRoot);
+    }
+
+    /// @inheritdoc IHubPortal
+    function setMerkleTreeBuilder(address merkleTreeBuilder_) external onlyOwner {
+        merkleTreeBuilder = merkleTreeBuilder_;
```

Comment:
Could be useful to check that the address being set is not address zero.
Even if it can be set at any time.